llama-index-core/llama_index/core/vector_stores/types.py

```diff
@@ -69,8 +68,10 @@ class FilterOperator(str, Enum):
     NE = "!="  # not equal to (string, int, float)
     GTE = ">="  # greater than or equal to (int, float)
     LTE = "<="  # less than or equal to (int, float)
-    IN = "in"  # metadata in value array (string or number)
-    NIN = "nin"  # metadata not in value array (string or number)
+    IN = "in"  # In array (string or number)
+    NIN = "nin"  # Not in array (string or number)
+    ANY = "any"  # Contains any (array of strings)
+    ALL = "all"  # Contains all (array of strings)
```

Comment:
@brenkehoe @logan-markewich Personally, I think the names "any" and "all" are too generic. Maybe "contains_any" and "contains_all" are more specific and easier to understand.

For example, Milvus also supports ["array_contains", "array_contains_any" and "array_contains_all"](https://milvus.io/docs/boolean.md).